engine/src/main/java/com/datatorrent/stram/client/AppPackage.java

```diff
@@ -329,6 +330,13 @@ public void processAppDirectory(boolean skipJars)
 
     Configuration config = new Configuration();
 
+    // load configuration with default properties
+    if (MapUtils.isNotEmpty(defaultProperties)) {
```

Comment:
The check is not needed as defaultProperties can't be null and it is OK to iterate over empty Map.